packages/json-api/src/-private/cache.ts

```diff
@@ -557,10 +557,12 @@ export default class JSONAPICache implements Cache {
       this._capabilities.notifyChange(identifier, 'state', null);
     }
 
+    const fields = this._capabilities.schema.fields(identifier);
```

Comment:
Technically I could only set this if one of the conditions below are met. Don't want to over-optimize